elastic/kibana
# [Time to Visualize] Embeddable Error Handling Via Output

## Summary

Fixes https://github.com/elastic/kibana/issues/81399.

This PR uses the `output$` Observable as described in https://github.com/elastic/kibana/issues/81399#issuecomment-714493318 to notify the embeddable panel when a fatal error has occurred. 

This PR also registers a factory for the ErrorEmbeddable to allow for easier control of when error embeddables appear. It also makes use of the existing `replacePanel` functionality in dashboard to replace any panel that encounters a fatal error with an error embeddable.

**This PR causes an issue where the user is able to save a dashboard with an ErrorEmbeddable present. In this case, the error embeddable cannot be replaced if the error is fixed. An alternative PR fixes this issue: https://github.com/elastic/kibana/pull/82201**

### How to Test

1. Create a new Lens Visualization and add it to the library
2. Add the lens vis to a dashboard and save the dashboard
3. Use the saved objects management section to delete the saved object associated with the lens visualization.
4. Reload the dashboard. You should see a 404 error.

### For maintainers
